waku/waku_core/time.nim

```diff
@@ -7,8 +7,12 @@ import std/times, metrics
 
 type Timestamp* = int64 # A nanosecond precision timestamp
 
-proc getNanosecondTime*(timeInSeconds: int64 | float64): Timestamp =
-  let ns = Timestamp(timeInSeconds * 1_000_000_000)
+proc getNanosecondTime*(timeInSeconds: int64): Timestamp =
+  let ns = Timestamp(timeInSeconds * int64(1_000_000_000))
+  return ns
+
+proc getNanosecondTime*(timeInSeconds: float64): Timestamp =
+  let ns = Timestamp(timeInSeconds * float64(1_000_000_000))
   return ns
```

Comment:
For some reason `proc getNanosecondTime*(timeInSeconds: int64 | float64): Timestamp` didn't work in Android x86. I think in practice these two functions I defined here have the same behavior :shrug: